Triangulation_2/include/CGAL/Triangulation_2_projection_traits_3.h

```diff
@@ -259,41 +302,69 @@ class Triangulation_2_projection_traits_3
 {
   typedef Triangulation_2_projection_traits_3<Kernel> Self;
 
-  typename Kernel::Vector_3 n;
+  typename Kernel::Vector_3 n, b1, b2;
 
 public:
   typedef typename Kernel::Vector_3 Vector_3;
 
 
-  Triangulation_2_projection_traits_3(const Vector_3& n_)
+  explicit Triangulation_2_projection_traits_3(const Vector_3& n_)
     : n(n_)
-  {}
+  {
+    typedef typename Kernel::FT FT;
+    typedef typename Kernel::Vector_3 Vector_3;
+
+    const FT& nx = n.x();
+    const FT& ny = n.y();
+    const FT& nz = n.z();
+    const FT anz = CGAL::abs(nz);
+    if(anz >= CGAL::abs(nx) &&
```

Comment:
Well, I am at the level of writing a traits class. That class template does not pretend to be a traits adapter, that would use the functors from another traits. Do you think it should be?